### PR TITLE
[Future Preview Environments] Terraform Certificates

### PR DESCRIPTION
**Pull request by vulkoingim**
## Description

Create cert-manager certificate resource using TF.

## Related Issue(s)

Fixes https://github.com/gitpod-io/ops/issues/5225

## Release Notes

```release-note
NONE
```

## Werft options:

- [ ] /werft with-local-preview
      If enabled this will build `install/preview`
- [x] /werft with-preview
- [ ] /werft with-integration-tests=all
      Valid options are `all`, `workspace`, `webapp`, `ide`

### CODE REVIEW

**Comment by roboquat** [comment]
@vulkoingim: Adding the "do-not-merge/release-note-label-needed" label because no release-note block was detected, please follow our [release note process](https://git.k8s.io/community/contributors/guide/release-notes.md) to remove it.

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository.
</details>

File: .werft/jobs/build/prepare.ts [hunk 1]
```diff
@@ -74,19 +72,9 @@ function configureStaticClustersAccess() {
...
-    werft.log(prepareSlices.ISSUE_CERTIFICATES, prepareSlices.ISSUE_CERTIFICATES);
-    var certReady = await issueMetaCerts(werft, certName, "certs", domain, config.repository.branch, prepareSlices.ISSUE_CERTIFICATES);
-    werft.done(prepareSlices.ISSUE_CERTIFICATES);
-    return certReady
-}
-
 async function decideHarvesterVMCreation(werft: Werft, config: JobConfig) {
     // always try to create - usually it will be no-op, but if tf changed for any reason we would reconcile
-    if (config.withPreview) {
+    if (config.withPreview && (!vmExists({ name: config.previewEnvironment.destname }) || config.cleanSlateDeployment)) {
```
Comment (vulkoingim) [inline .werft/jobs/build/prepare.ts:77]: Re-adding this check until all preview envs have the cert created with tf, so it doesn't fail on reapply.

File: .werft/util/certs.ts [hunk 2]
```diff
@@ -91,64 +79,6 @@ function retrieveFailedCertDebug(certName: string, slice: string) {
     );
 }
 
-function validateSubdomains(werft: Werft, slice: string, domain: string, subdomains: string[]): void {
```
Comment (vulkoingim) [inline .werft/util/certs.ts:94]: No need for this one as we cap the preview name to 20 chars and combined with the existing domain we never exceed 63 chars.

**Review by mads-hartmann — APPROVED** [review]
Preview environment certificate seems happy so 🚀 

Love that it's so much easier to understand what's going on - looking forward to 🪓 more of the TS code

**Pull request merged**